examples/usages/context_menu.rs

```diff
@@ -0,0 +1,201 @@
+//! This example illustrates how to create a context menu that changes the clear color
+
+use bevy::{
+    color::palettes::basic,
+    ecs::{relationship::RelatedSpawner, spawn::SpawnWith},
+    prelude::*,
+};
+use std::fmt::Debug;
+
+// event opening a new context menu at position `pos`
+#[derive(Event)]
+struct OpenContextMenu {
+    pos: Vec2,
+}
+
+// event will be send to close currently open context menus
+#[derive(Event)]
+struct CloseContextMenus;
+
+// marker component identifying root of a context menu
```

Comment:
nit: These could be doc comments 🤷  ; they are not really public, but rust-analyzer displays them so it sometimes help. (and then you can use doc links to avoid forgetting to rename them if somehow they change.

Reply:
good point. changed to doc comments